Source/Applications/Wave Demo Apps/UpdateWAVMetaData/Program.cs

```diff
@@ -109,12 +94,12 @@ static int Main(string[] args)
                     if (Convert.ToInt32(connection.ExecuteScalar(database.ParameterizedQueryString("SELECT COUNT(*) FROM Device WHERE Acronym = {0}", "acronym"), acronym)) == 0)
                     {
                         parameterizedQuery = database.ParameterizedQueryString("INSERT INTO Device(NodeID, Acronym, Name, ProtocolID, FramesPerSecond, " +
-                            "MeasurementReportingInterval, ConnectionString, Enabled) VALUES(" + nodeIDQueryString + ", {0}, {1}, {2}, {3}, {4}, {5}, {6})",
+                            "MeasurementReportingInterval, ConnectionString, Enabled) VALUES({0}, {1}, {2}, {3}, {4}, {5}, {6}, {7})",
                             "acronym", "name", "protocolID", "framesPerSecond", "measurementReportingInterval",
-                            "connectionString", "enabled");
+                            "connectionString", "enabled", "nodeID");
 
                         // Insert new device record
-                        connection.ExecuteNonQuery(parameterizedQuery, acronym, name, protocolID, sourceWave.SampleRate, 1000000, $"wavFileName={FilePath.GetAbsolutePath(sourceFileName)}; connectOnDemand=true; outputSourceIDs={acronym}; memoryCache={useMemoryCache}", database.Bool(true));
+                        connection.ExecuteNonQuery(parameterizedQuery, acronym, name, protocolID, sourceWave.SampleRate, 1000000, $"wavFileName={FilePath.GetAbsolutePath(sourceFileName)}; connectOnDemand=true; outputSourceIDs={acronym}; memoryCache={useMemoryCache}", database.Bool(true), database.Guid(nodeID)); memoryCache ={useMemoryCache}", database.Bool(true), nodeID);
```

Comment:
You need to use `database.Guid(nodeID)`. This is necessary because the ADO providers for SQLite, Oracle, and PostgreSQL all use the `string` data type instead of the `Guid` data type.

Reply:
Hmm, looking at this more closely, it looks like `database.Guid(nodeID)` is there, but there's something like a copy/paste error at the end of this line of code that needs to be removed.

Reply:
I've updated this. Again, my mistake.